### PR TITLE
Add profile picture

### PR DESCRIPTION
**Pull request by tanguoyang**
Add ProfilePicture to contact
A placeholder profile picture will be shown if theres no profile picture provided
A blank profilepicture will be shown if image address is unavailable

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -152,6 +153,8 @@ public static class EditPersonDescriptor {
         private Set<TechStack> techStack;
         private Set<Tag> tags;
 
+        private String profilePictureURL;
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:156]:
```suggestion
        private String profilePictureUrl;
```

Comment (maiyasaliha) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:156]: Coding Standard violation: variables in camel case

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -215,6 +219,10 @@ public Optional<GitHubUsername> getGitHubUsername() {
             return Optional.ofNullable(gitHubUsername);
         }
 
+        public void setProfilePictureURL(String url) {this.profilePictureURL = url; }
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:222]: Coding Standard violation: method names in camel case

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -215,6 +219,10 @@ public Optional<GitHubUsername> getGitHubUsername() {
...
+
+        public Optional<String> getProfilePictureURL() {return Optional.ofNullable(this.profilePictureURL);}
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:224]: Coding Standard violation: method names in camel case

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -82,6 +82,17 @@ public static Address parseAddress(String address) throws ParseException {
         return new Address(trimmedAddress);
     }
 
+    /**
+     * Parses a {@code String profilePictureURL} into an {@code String}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code address} is invalid.
+     */
+    public static String parseProfilePictureURL(String profilePictureURLArgument) {
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:91]: Coding Standard violation: method names in camel case

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -28,13 +28,16 @@ public class Person {
...
     private final Set<Tag> tags = new HashSet<>();
 
+    private String profilePictureURL;
+
     /**
      * Every field must be present and not null.
      */
     public Person(Name name, Phone phone, Email email, Address address, GitHubUsername gitHubUsername,
-                  Set<TechStack> techStack, Set<Tag> tags) {
+                  Set<TechStack> techStack, Set<Tag> tags, String profilePictureURL ) {
```
Comment (maiyasaliha) [inline src/main/java/seedu/address/model/person/Person.java:37]: Would it be better if we create a ProfilePicture Class instead?

**Review by maiyasaliha — DISMISSED** [review]
A few changes to be made, and build is failing. Once fixed we can merge!

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 6]
```diff
@@ -2,13 +2,7 @@
...
 import static java.util.Objects.requireNonNull;
 import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_GITHUB_USERNAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TECH_STACK;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (angsongyee) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:5]: Don't use wildcard imports

File: src/test/java/seedu/address/testutil/ContactBuilder.java [hunk 7]
```diff
@@ -3,12 +3,7 @@
 import java.util.HashSet;
 import java.util.Set;
 
-import seedu.address.model.contact.Address;
-import seedu.address.model.contact.Contact;
-import seedu.address.model.contact.Email;
-import seedu.address.model.contact.GitHubUsername;
-import seedu.address.model.contact.Name;
-import seedu.address.model.contact.Phone;
+import seedu.address.model.contact.*;
```
Comment (angsongyee) [inline src/test/java/seedu/address/testutil/ContactBuilder.java:6]: Same issue here

File: src/main/java/seedu/address/model/contact/Contact.java [hunk 8]
```diff
@@ -55,6 +60,8 @@ public Email getEmail() {
         return email;
     }
 
+    public ProfilePicture getProfilePicture() {return profilePicture; }
```
Comment (angsongyee) [inline src/main/java/seedu/address/model/contact/Contact.java:63]: I think a line break is needed here

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 9]
```diff
@@ -215,6 +223,10 @@ public Optional<GitHubUsername> getGitHubUsername() {
             return Optional.ofNullable(gitHubUsername);
         }
 
+        public void setProfilePicture(ProfilePicture profilePicture) {this.profilePicture = profilePicture; }
+
+        public Optional<ProfilePicture> getProfilePicture() {return Optional.ofNullable(this.profilePicture);}
```
Comment (angsongyee) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:228]: Needs line breaks

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 10]
```diff
@@ -166,14 +171,17 @@ public EditContactDescriptor(EditContactDescriptor toCopy) {
             setGitHubUsername(toCopy.gitHubUsername);
             setTechStack(toCopy.techStack);
             setTags(toCopy.tags);
+            setProfilePicture(toCopy.profilePicture);
         }
 
+        public EditContactDescriptor() {}
+
```
Comment (angsongyee) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:178]: Is there any usage for this default constructor?

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 11]
```diff
@@ -1,13 +1,7 @@
...
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_GITHUB_USERNAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_TECH_STACK;
+import static seedu.address.logic.parser.CliSyntax.*;
```
Comment (angsongyee) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:4]: Don't use wildcard imports

File: docs/UserGuide.md [hunk 12]
```diff
@@ -89,8 +89,8 @@ Format: `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS g/GITHUB_USERNAME [t/TAG] [
 
 Examples:
 
-- `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01 g/johnDoee`
-- `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 g/betBetty t/criminal ts/Flutter`
+- `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01 g/johnDoee pp/www.profilepicture.png`
+- `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 g/betBetty t/criminal ts/Flutter pp/www.profilepicture.png`
```
Comment (angsongyee) [inline docs/UserGuide.md:93]: `www.profilepicture.png` is an invalid input for ProfilePicture, and causes the app to crash

**Review by angsongyee — CHANGES_REQUESTED** [review]
image
I think the image could be smaller, and could be placed either before the Contact card on the left, or in the empty space on the right, instead of on top of the card

File: src/test/java/seedu/address/testutil/ContactBuilder.java (hunk 7 above)
Comment (tanguoyang) [reply src/test/java/seedu/address/testutil/ContactBuilder.java:6]: fixed

File: src/main/java/seedu/address/model/contact/Contact.java (hunk 8 above)
Comment (tanguoyang) [reply src/main/java/seedu/address/model/contact/Contact.java:63]: no need for a linebreak, following convention from existing code

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 9 above)
Comment (tanguoyang) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:228]: No need, am following convention set by existing code

File: docs/UserGuide.md (hunk 12 above)
Comment (tanguoyang) [reply docs/UserGuide.md]: fixed

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 11 above)
Comment (tanguoyang) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:4]: fixed

**Review by yuechen2001 — APPROVED** [review]

**Review by angsongyee — APPROVED** [review]

**Pull request merged**